### PR TITLE
delete user

### PR DESCRIPTION
**Pull request by phuongbk**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -47,4 +47,8 @@ group :development, :test do
 end
 
 # Use Bootstrap for stylesheets
-  gem "bootstrap-sass",       "3.2.0.0"
+  gem "will_paginate"
+  gem "bootstrap-will_paginate"
+
+#
```
Comment (phonghenry14) [inline Gemfile]: remove

File: app/helpers/sessions_helper.rb [hunk 2]
```diff
@@ -25,19 +26,23 @@ def logged_in?
...
-  def forget user
+  def forget(user)
     user.forget
     cookies.delete :user_id
     cookies.delete :remember_token
   end
 
   def log_out
-    forget current_user
+  	forget current_user
```
Comment (phonghenry14) [inline app/helpers/sessions_helper.rb]: indent

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -1,7 +1,8 @@
 module UsersHelper
-  def gravatar_for user
+  def gravatar_for (user, options = { size: Settings.avatar_size_default})
```
Comment (phonghenry14) [inline app/helpers/users_helper.rb]: gọn

File: app/views/admin/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<% provide(:title, "All users") %>
```
Comment (phonghenry14) [inline app/views/admin/users/index.html.erb]: remove (), i18n

File: app/views/admin/users/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+<h1>All users</h1>
```
Comment (phonghenry14) [inline app/views/admin/users/index.html.erb]: i18n

File: app/controllers/admin/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
+class Admin::UsersController < ApplicationController
+  before_action :logged_in_admin
+
+  def index
+    @users = User.normal.paginate page: params[:page], per_page: Settings.per_page
+  end
+
+  def destroy
+    User.find(params[:id]).destroy
```
Comment (leminhtuan2015) [inline app/controllers/admin/users_controller.rb]: Notice if cannot delete

File: app/helpers/sessions_helper.rb [hunk 7]
```diff
@@ -25,19 +26,23 @@ def logged_in?
...
+    current_user.role == Settings.admin
+  end
+
   def log_out
     session.delete :user_id
     @current_user = nil
   end
 
-  def forget user
+  def forget(user)
```
Comment (leminhtuan2015) [inline app/helpers/sessions_helper.rb]: remove ( )

File: app/views/layouts/_header.html.erb [hunk 8]
```diff
@@ -7,6 +7,16 @@
         <li><%= link_to t("title.help"), help_path %></li>
         <% if logged_in? %>
           <li><%= link_to t("title.user"), "#" %></li>
+          <% if admin? %>
+            <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: Link_to

File: app/controllers/admin/users_controller.rb [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    flash[:success] = t "flash.deleted"
+    redirect_to admin_users_url
+  end
+
+  private
+    def logged_in_admin
```
Comment (VoTaiTri) [inline app/controllers/admin/users_controller.rb]: indent

File: app/helpers/admin/users_helper.rb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+module Admin::UsersHelper
```
Comment (VoTaiTri) [inline app/helpers/admin/users_helper.rb]: remove file if useless

File: app/views/admin/users/_user.html.erb [hunk 11]
```diff
@@ -0,0 +1,10 @@
+<aside>
+    <section class="user_info">
```
Comment (VoTaiTri) [inline app/views/admin/users/_user.html.erb]: indent

File: config/routes.rb [hunk 12]
```diff
@@ -1,12 +1,17 @@
...
-  resources :users, except: [:index, :destroy]
+  get "help"      =>  "static_pages#help"
+  get "about"     =>  "static_pages#about"
+  get "contact"   =>  "static_pages#contact"
+  get "signup"    =>  "users#new"
+  
+  resources :users, expect: :destroy
+  
+  namespace :admin do
+    resources :users
```
Comment (VoTaiTri) [inline config/routes.rb]:
```ruby
only: ...

File: app/helpers/users_helper.rb (hunk 3 above)
Comment (VoTaiTri) [inline app/helpers/users_helper.rb]: indent, remove space after { and before }

File: app/views/layouts/_header.html.erb [hunk 13]
```diff
@@ -6,7 +6,16 @@
         <li><%= link_to t("title.home2"), root_path %></li>
         <li><%= link_to t("title.help"), help_path %></li>
         <% if logged_in? %>
-          <li><%= link_to t("title.user"), "#" %></li>
+          <% if admin? %>
+            <li class="dropdown">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: It is better use link_to

File: app/views/layouts/_header.html.erb [hunk 14]
```diff
@@ -6,11 +6,18 @@
         <li><%= link_to t("title.home2"), root_path %></li>
         <li><%= link_to t("title.help"), help_path %></li>
         <% if logged_in? %>
-          <li><%= link_to t("title.user"), "#" %></li>
+          <% if admin? %>
+            <li class="dropdown">
+            <%= link_to t("menu.mana"), "#mana_modal", :class => "dropdown-toggle",
+              :data => {:toggle=>"dropdown"} %> <b class="caret"></b>
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: Convention

File: app/views/layouts/_header.html.erb [hunk 15]
```diff
@@ -6,11 +6,18 @@
...
+              <li><%= link_to t("menu.mana_users"), admin_users_path %></li>
+            </ul>
+          </li>
+          <% end %>
           <li class="dropdown">
-            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
-              <% t "menu.account" %><b class="caret"></b>
-            </a>
+            <%= link_to t("menu.account"), "#account_modal", :class => "dropdown-toggle", 
+              :data => {:toggle=>"dropdown"} %><b class="caret"></b>
```
Comment (leminhtuan2015) [inline app/views/layouts/_header.html.erb]: Convention

File: app/controllers/sessions_controller.rb [hunk 16]
```diff
@@ -7,7 +7,11 @@ def create
     if user && user.authenticate(params[:session][:password])
       log_in user
       params[:session][:remember_me] == Settings.remember ? remember(user) : forget(user) 
-      redirect_to user
+      if user.role == Settings.admin
```
Comment (dieunb) [inline app/controllers/sessions_controller.rb]:
```
redirect_to user.role == Settings.admin ? admin_users_url : user
```

File: app/models/user.rb [hunk 17]
```diff
@@ -14,6 +14,9 @@ class User < ActiveRecord::Base
 
   before_save {self.email = email.downcase}
   
+  scope :normal, ->{where role: Settings.normal}
+  scope :is_admin, ->{where role: Settings.admin}
```
Comment (dieunb) [inline app/models/user.rb]: `admins`

File: app/models/user.rb [hunk 18]
```diff
@@ -14,6 +14,9 @@ class User < ActiveRecord::Base
 
   before_save {self.email = email.downcase}
   
+  scope :normal, ->{where role: Settings.normal}
```
Comment (dieunb) [inline app/models/user.rb]: normal

File: app/models/user.rb (hunk 17 above)
Comment (phuongbk) [inline app/models/user.rb]: is_admins hay Settings.admins hay đổi cả 2 vậy a?

**Pull request merged**